### PR TITLE
Feature/#15 lobby

### LINKED ISSUE #6: Extend the lobby functionalities
Display all players in the lobby
Selection of the figures
Leaving the lobby as a player
In the lobby of the game creator could be additional admin functions available like aborting the creation of the game

### LINKED ISSUE #11: Choose game piece
**Describe the solution you'd like**
In the lobby the player should be able to choose his game piece. Every game piece can only be chosen once. When a player has chosen no game piece or a game piece that is already taken he cannot press the ready button. After he pressed the ready button he can no longer change his game piece.

**Additional context**

### LINKED ISSUE #12: Players should be ready to start a game
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
Every client should have a ready button he can click when he is ready to start a game. The host can only start a game when every other player is ready.

**Additional context**

### LINKED ISSUE #15: Lobby
Create a screen where all players meet, which have already joined the game.

### PR DESCRIPTION
**Pull request by bferijan**
Closes #6 , Closes #15 , Closes #12 , Closes #11

### CODE REVIEW

File: app/src/main/java/com/ss20/se2/monopoly/models/Lobby.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+		boolean calculateReady = true;
+		if (players.size() >= 2 && players.size() <= NetworkUtilities.MAX_PLAYERS){
+			for (LobbyPlayer player : players){
+				if (!player.isHost() && !player.isReady()){
+					calculateReady = false;
+					break;
+				}
+				for (LobbyPlayer lobbyPlayer : players){
+					if (!lobbyPlayer.equals(player) && player.getGamePiece().getName().equals(lobbyPlayer.getGamePiece().getName())){
+						calculateReady = false;
```
Comment (dermutzh) [inline app/src/main/java/com/ss20/se2/monopoly/models/Lobby.java:129]: Add a break statement, like in Line 125, so we can terminate early. :)

File: app/src/main/java/com/ss20/se2/monopoly/network/client/GameController.java [hunk 2]
```diff
@@ -57,26 +57,28 @@ public void run(){
 				Log.d(NetworkUtilities.TAG, "Client cant connect because Server full");
 			}else if (message.equals("ok")){
 				communicator = new ClientToServerCommunicator(socket);
+				joined = true;
```
Comment (dermutzh) [inline app/src/main/java/com/ss20/se2/monopoly/network/client/GameController.java]: These if statements are a perfect situation for a switch case. I would recommend replacing it.

File: app/src/main/java/com/ss20/se2/monopoly/view/ClientLobbyFragment.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+			readymessage.setSenderAddress(Lobby.getInstance().getSelf().getAddress());
+			readymessage.setSenderPort(Lobby.getInstance().getSelf().getPort());
+			readymessage.setType(RequestType.JOIN_GAME);
+			GameController.getInstance().changeReadyLobby(readymessage);
+		}else if (id == R.id.backBtn){
+			LeaveLobbyNetworkMessage message = new LeaveLobbyNetworkMessage();
+			message.setSenderName(Lobby.getInstance().getSelf().getName());
+			message.setSenderAddress(Lobby.getInstance().getSelf().getAddress());
+			message.setSenderPort(Lobby.getInstance().getSelf().getPort());
+			message.setType(RequestType.JOIN_GAME);
```
Comment (dermutzh) [inline app/src/main/java/com/ss20/se2/monopoly/view/ClientLobbyFragment.java]:
```suggestion
			message.setType(RequestType.LEAVE_GAME);
```

**Review by dermutzh — CHANGES_REQUESTED** [review]
Looks good so far, please take a look at the three small issues. After that feel free to merge 😊

File: app/src/main/java/com/ss20/se2/monopoly/models/Lobby.java (hunk 1 above)
Comment (bferijan) [reply app/src/main/java/com/ss20/se2/monopoly/models/Lobby.java:129]: Ok done. 
Added another break to improve performance.

File: app/src/main/java/com/ss20/se2/monopoly/view/ClientLobbyFragment.java (hunk 3 above)
Comment (bferijan) [reply app/src/main/java/com/ss20/se2/monopoly/view/ClientLobbyFragment.java]: Thanks, done :)

**Pull request merged**